File: book_kit/src/main/java/net/lzbook/kit/utils/statistic/alilog/LogGroup.java
package net.lzbook.kit.utils.statistic.alilog;

import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Created by wangjwchn on 16/8/2.
 */

public class LogGroup {
    protected List<Log> mContent = new ArrayList<Log>();
    private String mTopic = "";
    private String mSource = "";

    public LogGroup() {
    }

    public LogGroup(String topic, String source) {
        mTopic = topic;
        mSource = source;
    }

    public void PutTopic(String topic) {
        mTopic = topic;
    }

    public void PutSource(String source) {
        mSource = source;
    }

    public void PutLog(Log log) {
        mContent.add(log);
    }

    public String LogGroupToJsonString() {
        JSONObject json_log_group = new JSONObject();
        json_log_group.put("__source__", mSource);
        json_log_group.put("__topic__", mTopic);
        JSONArray log_arrays = new JSONArray();

        for (Log log : mContent) {
            Map<String, Object> map = log.GetContent();
            JSONObject json_log = new JSONObject(map);
            log_arrays.add(json_log);
        }
        json_log_group.put("__logs__", log_arrays);
        String s = json_log_group.toJSONString();
        return s;
    }

}

File: dy-statistics-sdk/src/main/java/com/dingyue/statistics/log/LogGroup.kt
package com.dingyue.statistics.log

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

/**
 * Created by wangjwchn on 16/8/2.
 */

class LogGroup {
    private var mContent: MutableList<ServerLog> = ArrayList()
    private var mTopic = ""
    private var mSource = ""
    var project: String? = null
    var logstore: String? = null

    val logs: List<ServerLog>
        get() = mContent

    constructor(topic: String, source: String, project: String, logstore: String) {
        mTopic = topic
        mSource = source
        this.project = project
        this.logstore = logstore
    }

    constructor(topic: String, source: String) {
        mTopic = topic
        mSource = source
    }

    fun putTopic(topic: String) {
        mTopic = topic
    }

    fun putSource(source: String) {
        mSource = source
    }

    fun putLog(log: ServerLog) {
        mContent.add(log)
    }

    fun toJsonString(): String? {
        try {
            val json_log_group = JSONObject()
            json_log_group.put("__source__", mSource)
            json_log_group.put("__topic__", mTopic)
            val log_arrays = JSONArray()

            for (log in mContent) {
                val map = log.content
                val json_log = JSONObject(map)
                log_arrays.put(json_log)
            }
            json_log_group.put("__logs__", log_arrays)
            return json_log_group.toString()
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return null
    }

}
